tests/microos/install_image.pm

```diff
@@ -0,0 +1,100 @@
+use power_action_utils 'power_action';
+use serial_terminal 'select_serial_terminal';
+
+sub run {
+    select_serial_terminal;
+
+    # Use image name from HDD_1 variable
+    my $image = get_required_var('HDD_1');
+    # Use sda as target disk by default
+    my $device = get_var("MICRO_INSTALL_IMAGE_TARGET_DEVICE", "/dev/sda");
+    # Use partition prefix for nvme devices
+    my $prefix = $device =~ /nvme/ ? "p" : "";
+    # SL Micro x86_64 image has three partitions, aarch64 and ppc64le images have only two partitions
+    my $root_partition_id = is_x86_64 ? 3 : 2;
+    my $root_partition = "${device}" . $prefix . $root_partition_id;
+    # New partition id for ignition will be directly after root partition
+    my $ignition_partition_id = $root_partition_id + 1;
+    my $ignition_partition = "${device}" . $prefix . $ignition_partition_id;
+    record_info("Device information", "Device: ${device}\nRoot partition: ${root_partition}\nIgnition partition: ${ignition_partition}");
+
+    # Mount nfs share with images
+    assert_script_run("mount -o ro,noauto,nofail,nolock -t nfs openqa.suse.de:/var/lib/openqa/share /mnt");
+    assert_script_run("ls -als /mnt/factory/hdd/${image}");
+    # dd image to disk
+    assert_script_run("xzcat /mnt/factory/hdd/${image} | dd of=${device} bs=65536 status=progress", timeout => 300);
+    assert_script_run("sync");
+    assert_script_run("umount /mnt");
+
+    my $device_layout = script_output("lsblk");
+    record_info("Device layout", ${device_layout});
+
+    # Modify disk to be able to correctly boot and login
+    assert_script_run("mount ${root_partition} /mnt");
+    assert_script_run("btrfs property set /mnt ro false");
+    # Set correct serial console to be able to see login in first boot
+    assert_script_run("sed -i 's/console=ttyS0,115200/console=ttyS1,115200/g' /mnt/boot/grub2/grub.cfg") if is_x86_64;
+    # Upload original grub configuration
+    upload_logs("/mnt/etc/default/grub", failok => 1);
+    # Set permanent grub configuration
+    assert_script_run("sed -i 's/console=ttyS0,115200/console=ttyS1,115200/g' /mnt/etc/default/grub") if is_x86_64;
+    # Fully disable graphical terminal on legacy systems without UEFI
+    assert_script_run("sed -i 's/#GRUB_TERMINAL=console/GRUB_TERMINAL=console/g' /mnt/etc/default/grub") if (is_ipmi && !get_var('IPXE_UEFI'));
+    # We need to properly set grub terminal bsc#1230844, otherwise grub menu will not be visible in non graphical boot environments
+    assert_script_run("sed -i 's/GRUB_TERMINAL_INPUT=\".*\"/GRUB_TERMINAL_INPUT=\"console gfxterm\"/g' /mnt/etc/default/grub");
+    assert_script_run("sed -i 's/GRUB_TERMINAL_OUTPUT=\".*\"/GRUB_TERMINAL_OUTPUT=\"console gfxterm\"/g' /mnt/etc/default/grub");
+    # Enable root loging with password
+    assert_script_run("echo 'PermitRootLogin yes' > /mnt/etc/ssh/sshd_config.d/root.conf");
+    assert_script_run("btrfs property set /mnt ro true");
+    assert_script_run("umount /mnt");
```

Comment:
How about adding support of combustion script? Then these changes can be easily done in that and it will be easier to extend/customized in future based on user's need?

Reply:
I have no experience with combustion, and I believe that for these simple tasks, we don't need such complexity right from the start. I will create follow where we can discuss additional features.